### ISSUE TITLE
Calendar App disabled and fails to re-enable

### ISSUE THREAD

**Issue by jurgenhaas** [nextcloud/calendar#2054]
NextCloud version: 18.0.1

Calendar App version: 2.0.2

When re-enabling the app, the icon doesn't appear and the error log shows

```
{"reqId":"YWNGDCTZauB2D5bb9lNr","level":3,"time":"2020-03-06T07:59:30+00:00","remoteAddr":"217.8.58.12","user":"jurgenhaas","app":"no app in context","method":"GET","url":"/index.php/apps/files","message":{"Exception":"Error","Message":"Call to undefined method OCA\\Calendar\\AppInfo\\Application::registerNavigation()","Code":0,"Trace":[{"file":"/var/www/nextcloud/bitegra/lib/private/legacy/app.php","line":266,"function":"require_once"},{"file":"/var/www/nextcloud/bitegra/lib/private/legacy/app.php","line":155,"function":"requireAppFile","class":"OC_App","type":"::","args":[{"__class__":"OCA\\Calendar\\AppInfo\\Application"}]},{"file":"/var/www/nextcloud/bitegra/lib/private/legacy/app.php","line":128,"function":"loadApp","class":"OC_App","type":"::","args":["calendar"]},{"file":"/var/www/nextcloud/bitegra/lib/base.php","line":993,"function":"loadApps","class":"OC_App","type":"::","args":[]},{"file":"/var/www/nextcloud/bitegra/index.php","line":38,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/nextcloud/bitegra/apps/calendar/appinfo/app.php","Line":26,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:74.0) Gecko/20100101 Firefox/74.0","version":"18.0.1.3","id":"5e62030371269"}
```

When going to system settings overview, all is green and no outstanding maintenance is required. So what could be causing this?

**Comment by tcitworld** [comment]
Remove the app completely and reinstall it?

**Comment by jurgenhaas** [comment]
Am I losing data with this approach?

**Comment by tcitworld** [comment]
Sorry about the misclick.

No, the Calendar app is only a CalDAV client running in your browser, data is stored elsewhere (and deleting an app never actually deletes database entries anyway).

**Comment by jurgenhaas** [comment]
Confirmed, I have removed the code and re-downloaded the release from github, now it's working correctly. Just wondering how that came about? Have I probably downloaded the release too early and it was corrected afterwards? I'm maintaining a dozen NC installations and have automated processes in place that update every app automatically every night. Never had a problem with that for the last 12 months.
